### PR TITLE
CFE-2898: New data validation functions

### PR DESCRIPTION
**Pull request by karlhto**
validdata and validjson are now implemented as functions to make sure that data files can be parsed by CFEngine's JSON parser. Functionality for YAML, CSV and ENVFILE is expected to be added eventually.

Ticket: CFE-2898
Changelog: Title

### CODE REVIEW

**Comment by karlhto** [comment]
It seems that adding functionality for `csv` and `envfile` will require some additional functionality being implemented in `libntech`. I think it will be a better idea to split that into a separate PR. :)

**Comment by karlhto** [comment]
@cf-bottom Jenkins please

**Comment by cf-bottom** [comment]
Sure, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/4514/)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/4514/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-4514/

**Comment by karlhto** [comment]
Cool false positives @lgtm-com!

**Comment by karlhto** [comment]
@cf-bottom Jenkins

**Comment by cf-bottom** [comment]
Sure, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/4550/)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/4550/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-4550/

**Comment by karlhto** [comment]
@cf-bottom Jenkins please

**Comment by cf-bottom** [comment]
Sure, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/4594/)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/4594/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-4594/

File: libpromises/evalfunction.c [hunk 1]
```diff
@@ -9015,6 +9082,13 @@ static const FnCallArg READDATA_ARGS[] =
     {NULL, CF_DATA_TYPE_NONE, NULL}
 };
 
+static const FnCallArg VALIDDATA_ARGS[] =
+{
+    {CF_ABSPATHRANGE, CF_DATA_TYPE_STRING, "Data to validate"},
+    {"YAML,JSON", CF_DATA_TYPE_OPTION, "Type of data to validate"},
```
Comment (vpodzime) [inline libpromises/evalfunction.c]: Should only mention `"JSON"` if that's the only supported format for now.

File: libpromises/evalfunction.c [hunk 2]
```diff
@@ -8966,6 +9027,12 @@ static const FnCallArg READFILE_ARGS[] =
     {NULL, CF_DATA_TYPE_NONE, NULL}
 };
 
+static const FnCallArg VALIDDATATYPE_ARGS[] =
+{
+    {CF_ABSPATHRANGE, CF_DATA_TYPE_STRING, "Data to validate"},
```
Comment (olehermanse) [inline libpromises/evalfunction.c]: Seems like this should be `CF_ANYSTRING` range, not `CF_ABSPATHRANGE`.

File: libpromises/evalfunction.c [hunk 3]
```diff
@@ -9015,6 +9082,13 @@ static const FnCallArg READDATA_ARGS[] =
     {NULL, CF_DATA_TYPE_NONE, NULL}
 };
 
+static const FnCallArg VALIDDATA_ARGS[] =
+{
+    {CF_ABSPATHRANGE, CF_DATA_TYPE_STRING, "Data to validate"},
```
Comment (olehermanse) [inline libpromises/evalfunction.c]: Seems like this should be `CF_ANYSTRING` range, not `CF_ABSPATHRANGE`.

**Comment by karlhto** [comment]
Only requested changes made, so another jenkins run is unnecessary.

**Pull request merged**